lookout/style/typos/benchmarks/typo_commits_report.py

```diff
@@ -0,0 +1,193 @@
+        wrong_identifier = dataset_row["wrong id"]
+        for fix in fixes:
+            if fix.identifier == wrong_identifier:
+                if correct_fix in fix.identifier_candidates:
+                    metrics.detection_true_positive += 1
+                    metrics.fix_accuracy += 1
+            else:
+                metrics.detection_false_positive += 1
+        return json.dumps(metrics.to_dict())
+
+    def _trigger_review_event(self, dataset_row: Dict[str, Any]) -> Sequence[TypoFix]:
+        comments = self._analyzer_context_manager.review(
+            dataset_row["commit"], "HEAD", git_dir=dataset_row["repo"], bblfsh=self._bblfsh,
+            log_level="info", config_json=self._config)
+        typo_fixes = []
+        for comment in comments:
+            typo_fix_dict = json.loads(comment.text)
+            typo_fixes.append(TypoFix(head_file=typo_fix_dict["head_file"],
+                                      line_number=typo_fix_dict["line_number"],
+                                      candidates=typo_fix_dict["candidates"],
+                                      token=typo_fix_dict["token"]))
+        return typo_fixes
+
+    def _finalize(self, reports: Iterable[Dict[str, str]]) -> Iterator[Dict[str, str]]:
+        """
+        Summarize all individual reports to the final one.
+
+        :param reports: Reports generated by `TypoCommitsReporter.generate_commit_dataset_report()`
+        :return: Summarized final report
+        """
+        # TODO(zurk): Add a proper report
+        sum_metric = self.get_metrics_billet()
+        for report in reports:
+            sum_metric += pandas.Series(json.loads(report)["report"])
+        yield json.dumps(sum_metric.to_dict())
+
+    @staticmethod
+    def get_metrics_billet() -> pandas.Series:
+        """
+        Generate pandas series with TypoCommitsReporter metrics.
+
+        `detection_` prefix relates metric to typo detection and `fix_` to a metrics for founded
+        typos. Support is a number of analyzed identifiers.
+        """
+        metrics = ((
+            ("detection_true_positive", 0.0),
+            ("detection_false_positive", 0.0),
+            ("fix_accuracy", 0.0),
+            ("support", 0.0),
+        ))
+        index, defaults = zip(*metrics)
+        return pandas.Series(data=defaults, index=index)
+
+
+def generate_typos_report_entry(dataset: str, output: str, bblfsh: str, config: dict,
+                                database: Optional[str] = None, fs: Optional[str] = None,
+                                repos_cache: Optional[str] = None) -> None:
+    """
+    Entry point for command line interface to generate typos quality report for the given data.
+
+    :param dataset: csv file with commits to make report. Should contain repo, commit, file, \
+                    line, wrong id and correct id columns.
+    :param output: Directory where to save report.
+    :param bblfsh: bblfsh address to use by lookout-sdk.
```

Comment:
```suggestion
    :param bblfsh: bblfsh address to use for `lookout-sdk`.
```

Reply:
btw, this often aliases with `import bblfsh`, I would rather have `bblfsh_addr` or smth